File: sample-videochat-webrtc/sample-videochat-webrtc/Components/CallViewController/StatsView.m
//
//  StatsView.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

#import "StatsView.h"

static NSString * const kStatsReportPlaceholderText = @"Loading stats report...";

@implementation StatsView {
    UILabel *_statsLabel;
}

- (instancetype)initWithFrame:(CGRect)frame {
    
    self = [super initWithFrame:frame];
    if (self != nil) {
        
        _statsLabel = [[UILabel alloc] initWithFrame:frame];
        _statsLabel.text = kStatsReportPlaceholderText;
        _statsLabel.numberOfLines = 0;
        _statsLabel.font = [UIFont fontWithName:@"Roboto" size:12];
        _statsLabel.adjustsFontSizeToFitWidth = YES;
        _statsLabel.minimumScaleFactor = 0.6;
        _statsLabel.textColor = [UIColor greenColor];
        [self addSubview:_statsLabel];
        self.backgroundColor = [UIColor colorWithWhite:0 alpha:.6];
    }
    return self;
}

- (void)setStats:(NSString *)stats {
    if (stats == nil) {
        stats = kStatsReportPlaceholderText;
    }
    _statsLabel.text = stats;
}

- (void)layoutSubviews {
    _statsLabel.frame = self.bounds;
}

- (void)setHidden:(BOOL)hidden {
    
    if (hidden) {
        
        [self setStats:nil];
    }
    
    [super setHidden:hidden];
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Call/Views/StatsView.swift
//
//  StatsView.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 12/18/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

struct StatsViewConstants {
    static let statsReportPlaceholderText = "Loading stats report..."
}

class StatsView: UIView {
    //MARK: - Properties
    lazy private var statsLabel: UILabel = {
        let statsLabel = UILabel(frame: bounds)
        statsLabel.text = StatsViewConstants.statsReportPlaceholderText
        statsLabel.numberOfLines = 0
        statsLabel.font = UIFont(name: "Roboto", size: 12.0)
        statsLabel.adjustsFontSizeToFitWidth = true
        statsLabel.minimumScaleFactor = 0.6
        statsLabel.textColor = UIColor.green
        return statsLabel
    }()
    
    override var isHidden: Bool {
        didSet {
            if isHidden == true {
                updateStats(nil)
            }
        }
    }
    
    //MARK: - Life Cycle
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        addSubview(statsLabel)
        backgroundColor = UIColor(white: 0.0, alpha: 0.6)
    }
    
    override func layoutSubviews() {
        statsLabel.frame = bounds
    }
    
    //MARK: - Internal Methods
    func updateStats(_ stats: String?) {
        statsLabel.text = stats ?? StatsViewConstants.statsReportPlaceholderText
    }
}


